### ISSUE TITLE
Centering tiles in TileContainer when only 2 tiles present

### ISSUE THREAD

**Issue by CantonCode** [marcoromag/react-tiles-dnd#1]
Hi,

First off, great library and love how it works.. One question/issue.. 

I want to have the width of the TileContainer 100% But i only want 2 tiles currently that takes up 50 percent of the entire container. Because of this the tiles fill only half the space and they keep left. 

I have tried aligning the content to center etc etc but cant seems to get the tiles into the middle of the tile container.

For example if i also only want 3 tiles I want these tiles to be centered in the container rather than forced left. Or if I have a full row of 4 tiles on top and 2 tiles on the bottom I want the two tiles on the bottom to be centered and not forced left. 

Does this make sense? let me know? I would be interested in contributing to making this adjustment also :) but maybe it is possible already and I am missing soemthing

**Comment by marcoromag** [comment]
Hi, Thanks for your suggestions.
From my understanding, you want the tiles to align to the center, and not to the left.
This poses a couple of issues because right now the position of each tile is calculated programmatically, and the library constructs a grid matrix to make the calculations.
All the calculations are done by taking the mouse position during the drag, and dividing the X axis by tileWidth and Y axis by tileHeight. In this way, the coordinates are translated back into 'grid' coordinates.
This approach works only when the tiles are effectively aligned with the grid (i.e. the X and Y position of each tile is a multiple of `tileWidth` and `tileHeight`), centering them would require a different approach, because the X position would not be a multiple of `tileWidth`.

In principle, it would be possible to implement a new StrategyInterface which can do what you need, but it would be slightly hard to make it.

What could be done is that the grid is still calculated as left-aligned, BUT if the row on the table is not complete, then we should consider an offset to convert the mouse X position to the grid position
Example:

suppose I have a 4x4 grid, and I have 2 1x1 table (1, 2 and 3) and a 2x2 tile (4), we would have a situation like
```
1 2 3 -
4 4 - -
4 4 - -
- - - -
```

but in reality, 1, 2 and 3 are centered in the row, and 4 is also centered.
When you start dragging, you need to consider that the position of 3 is actually offset by `2 * tileWidth / 2` (the space of 2 empty grid slot, half on each side).

to determine if (4) hovers 1,2, or 3, you also have to consider that those elements are offset by `1 * tileWidth /2`(the space of 1 empty grid slot, half on each side).

Bottom line: yes can be done, but we need to rethink completely the strategy logic.

The original idea of the library was to do something very similar to what iPhone does with the apps on the Springboard. But... with variable tile dimensions :)
